doc/esmvalcore/preprocessor.rst

```diff
@@ -138,344 +260,559 @@ losing generality. To mask ice out, `mask_landseaice` can be used:
       mask_landseaice:
         mask_out: ice
 
-and requires only one argument:
-* mask_out: either landsea or ice.
+and requires only one argument: ``mask_out``: either ``landsea`` or ``ice``.
 
-As in the case of `mask_landsea`, the preprocessor automatically retrieves the
-`fx: sftgif` mask.
+As in the case of ``mask_landsea``, the preprocessor automatically retrieves
+the ``fx_files: [sftgif]`` mask.
 
-Another option is to just read the fx masks as any other CMOR variable and use
-it within a diagnostic script. This can be done in the variable dictionary by
-specifiying the desired fx variables (masks):
+See also :func:`esmvalcore.preprocessor.mask_landseaice`.
 
-.. warning::
-  Code snippet, text and link to function to be added (after #1037 and #1075
-  are closed).
+Mask files
+----------
+
+At the core of the land/sea/ice masking in the preprocessor are the mask files
+(whether it be fx type or Natural Earth type of files); these files (bar
+Natural Earth) can be retrived and used in the diagnostic phase as well or
+solely. By specifying the ``fx_files:`` key in the variable in diagnostic in
```

Comment:
Some words missing after "solely"? Like "solely in the preprocessor"?

Reply:
Hmm, could also be that the "." is  wrong and the sentence should read "... as well or solely by specifying..."

Reply:
I think "solely" can be removed here.